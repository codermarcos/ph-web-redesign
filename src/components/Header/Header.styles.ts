import styled from 'styled-components';

import { border, padding, viewWidth } from 'csx';

import Link from 'next/link';

import { colors, sizes } from '@/styles/variables';

export const HeaderContainer = styled.header({
  backgroundColor: colors.whiteDefault,
  display: 'grid',
  gridTemplateAreas: '"marginL header marginR" "marginL navbar marginR"',
  gridTemplateColumns: `1fr minmax(auto, ${sizes.container}) 1fr`,
  gridTemplateRows: 'auto auto',
  justifyContent: 'center',
  padding: padding(
    sizes.space.default,
    sizes.space.default,
    0,
    sizes.space.default,
  ),
  width: viewWidth(100),
});

export const LoginLink = styled(Link)({
  alignSelf: 'center',
  backgroundColor: colors.accent,
  border: border({
    color: colors.accentDarkness1,
    style: 'solid',
    width: sizes.border.xxs,
  }),
  borderRadius: sizes.borderRadius.default,
  color: colors.whiteDefault,
  cursor: 'pointer',
  gridArea: 'header',
  justifySelf: 'center',
  padding: padding(sizes.space.default),
  textDecoration: 'none',

  // eslint-disable-next-line sort-keys
  ':visited': {
    color: colors.whiteDefault,
  },
});
