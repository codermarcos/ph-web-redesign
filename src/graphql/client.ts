import { ApolloClient } from '@apollo/client';

import cache from '@/graphql/cache';

const apolloClient = new ApolloClient({
  cache,
  uri: '/api/graphql',
});

export default apolloClient;
