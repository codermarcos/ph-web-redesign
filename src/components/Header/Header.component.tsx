import dynamic from 'next/dynamic';

import { useQuery } from '@apollo/client';

import {
  HeaderContainer,
  LoginLink,
} from '@/components/Header/Header.styles';

import NavBar from '@/components/NavBar/NavBar.component';

import QUERY_VIEWER from '@/graphql/queries/Viewer.graphql';

const LazyLoggedHeader = dynamic(
  () => import('@/components/LoggedHeader/LoggedHeader.component'),
);

function Header() {

  const { data } = useQuery(QUERY_VIEWER);

  return (
    <HeaderContainer>
      {
        data && data.viewer
          ? (<LazyLoggedHeader {...data.viewer.user} />)
          : (
            <LoginLink
              data-test-id="button-login"
              href="/api/authorize"
            >
              Log in with Product Hunt
            </LoginLink>
          )
      }

      <NavBar />
    </HeaderContainer>
  );

}

export default Header;
