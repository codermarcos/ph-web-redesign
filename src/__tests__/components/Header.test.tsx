import { RenderResult, render } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import { faker } from '@faker-js/faker';

import QUERY_VIEWER from '@/graphql/queries/Viewer.graphql';

import Header from '@/components/Header/Header.component';

faker.seed(1);

const user = {
  name: faker.name.fullName(),
  profilePhoto: faker.image.avatar(),
};

const mocks = [
  {
    request: {
      query: QUERY_VIEWER,
    },
    result: {
      data: {
        viewer: { user },
      },
    },
  },
];

describe(
  '[Component] -> components/Header',
  () => {

    // eslint-disable-next-line init-declarations
    let wrapper: RenderResult;

    beforeEach(
      () => {

        wrapper = render(
          <MockedProvider mocks={mocks}>
            <Header />
          </MockedProvider>
        );

      }
    );

    it(
      'should render correct header',
      () => {

        expect(wrapper.container).toMatchSnapshot();

      }
    );

  }
);
