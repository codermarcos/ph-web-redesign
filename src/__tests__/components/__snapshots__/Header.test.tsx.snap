// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] -> components/Header should render correct header 1`] = `
<div>
  <header
    class="sc-e1990ee5-0 cvCnGB"
  >
    <a
      class="sc-e1990ee5-1 kNGFan"
      href="/api/authorize"
    >
      Log in with Product Hunt
    </a>
  </header>
</div>
`;
